src/main/java/com/shapesecurity/shift/es2017/scope/ScopeAnalyzer.java

```diff
@@ -484,11 +525,15 @@ private State finish(@Nonnull Node astNode, @Nonnull Scope.Type scopeType, boole
                 case ParameterExpression:
                     // resolve only block-scoped free declarations
                     ImmutableList<Variable> variables3 = variables;
-                    for (Pair<String, ImmutableList<Declaration>> entry2 : this.blockScopedDeclarations.merge(this.functionDeclarations, ImmutableList::append).entries()) {
+                    ImmutableList<Pair<String, ImmutableList<Declaration>>> declarationEntries = this.blockScopedDeclarations.merge(this.functionDeclarations, ImmutableList::append).entries();
+                    List<Pair<String, ImmutableList<Declaration>>> declarationEntriesSorted = StreamSupport.stream(Spliterators.spliteratorUnknownSize(declarationEntries.iterator(), Spliterator.ORDERED), false)
```

Comment:
It might be cleaner to dedupe the three places this pattern occurs (`List declarationEntriesSorted = ...; for ...;` into a static method.